docs/proposals/image-updater-first-class-support.md

```diff
@@ -0,0 +1,239 @@
+2. `.status.summary.images` - Currently this field exists as a slice of strings that just contains the full qualified paths of all images associated to an application. This filed could be repurposed by changing it from `[]string` to `[]imageUpdate` to store the additional information above. This seems like a naturally better place for it, however this would be a breaking change (and at present  would also break Image Updater as it uses this field to get the list of potentially updateable images)
+3. `.status.summary.imageUpdates` - This field could instead be introduced within the applicaiton summary with the understanding that `status.summary.images` will be deprecated and eventually removed, since it will be rendered obsolete by this new field. 
+
+## Open Questions [optional]
+
+Based on this proposal, the Image Updater controller would need to write to the application status fields in order to track the state of image updates for various images. This would mean we could potentially have 2 controllers writing to the application resource. Since the introduction of server side apply in Argo CD, it is not uncommon to have multiple field managers controlling different fields in a resource, however it may be the case that this does not extend to the status fields. In such an event, should just the responsibility of updating the application status field for image updates be shifted to the application-controller? 
+
+
+### Security Considerations
+
+With the rise in interest in software supply chain security across the industry, one aspect worth considering is that the introduction of Image Updater into Argo CD would expose Argo CD toward an entirely new type of source repository in image registries. Image Updater interfaces directly with various image registries, and as such pulls image manifests from these sources, currently trusting that the user has configured reputable image registries for their purposes. Moving forward, it would be reasonable to suggest that we could be more careful and selective about which images/registries we would want to allow Argo CD to pull images from, in order to minimize exposure to bad actors. 
+
+One of the potential enhancements that could be introduced within Image Updater down the line would be the ability to verify that images are signed through potentially leveraging existing container verification solutions. The above proposed CR fields could be expanded to build out `.spec.image.verification`, for instance. This could be used to store specific configuration/constraints expressed by the user to ensure the trustworthiness of containers being used in their applications. One example of this would be storing the location of a public key to verify an image against.
+
```

Comment:
If we find a way to avoid having to push back to git, it would help to mitigate several security concerns.

Reply:
The points mentioned here are not related to git write back though

what would your main concern be with that?